eva-accession-release/src/test/java/uk/ac/ebi/eva/accession/release/io/VariantContextWriterTest.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    private Variant buildVariant(String chr1, int start, String reference, String alternate, String sequenceOntology,
+                                 String... studies) {
+        Variant variant = new Variant(chr1, start, start + alternate.length(), reference, alternate);
+        variant.setMainId(ID);
+        for (String study : studies) {
+            VariantSourceEntry sourceEntry = new VariantSourceEntry(study, FILE_ID);
+            sourceEntry.addAttribute(VARIANT_CLASS_KEY, sequenceOntology);
+            sourceEntry.addAttribute(STUDY_ID_KEY, study);
+            variant.addSourceEntry(sourceEntry);
+        }
+        return variant;
+    }
+
+    public void assertWriteVcf(File output, Variant... variants) throws Exception {
+        VariantContextWriter writer = new VariantContextWriter(output, REFERENCE_ASSEMBLY);
+        writer.open(null);
+
+        VariantToVariantContextProcessor variantToVariantContextProcessor = new VariantToVariantContextProcessor();
+        List<VariantContext> variantContexts =
+                Stream.of(variants).map(variantToVariantContextProcessor::process).collect(Collectors.toList());
+        writer.write(variantContexts);
+
+        writer.close();
+
+        assertTrue(output.exists());
+    }
+
+    @Test
+    public void checkReference() throws Exception {
+        File output = temporaryFolder.newFile("test.vcf");
+        assertWriteVcf(output, buildVariant(CHR_1, 1000, "C", "A", SNP_SEQUENCE_ONTOLOGY, STUDY_1));
+
+        List<String> referenceLines = grepFile(output, "reference");
+        assertEquals(1, referenceLines.size());
+        assertEquals("##reference=" + REFERENCE_ASSEMBLY, referenceLines.get(0));
+    }
+
+    private List<String> grepFile(File file, String contains) throws IOException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(file)));
+        List<String> lines = new ArrayList<>();
+        String line;
+        while ((line = reader.readLine()) != null) {
+            if (line.contains(contains)) {
+                lines.add(line);
+            }
+        }
+        reader.close();
+        return lines;
+    }
+
+    @Test
+    public void checkMetadataSection() throws Exception {
+        File output = temporaryFolder.newFile("test.vcf");
+        assertWriteVcf(output, buildVariant(CHR_1, 1000, "C", "A", SNP_SEQUENCE_ONTOLOGY, STUDY_1));
+
+        List<String> metadataLines = grepFile(output, "##");
+        assertEquals(4, metadataLines.size());
+        List<String> headerLines = grepFile(output, "#CHROM");
+        assertEquals(1, headerLines.size());
+    }
+
+    @Test
+    public void checkAccession() throws Exception {
+        File output = temporaryFolder.newFile("test.vcf");
+        assertWriteVcf(output, buildVariant(CHR_1, 1000, "C", "A", SNP_SEQUENCE_ONTOLOGY, STUDY_1));
+
+        List<String> dataLines = grepFile(output, ID);
```

Comment:
Could you add a test for a multi-allelic RS?

Reply:
Can't do a proper multiallelic, only several variants with the same RS and position